### PR TITLE
GCB-50: fetch data from Google API

### PR DESCRIPTION
**Pull request by michallaskowski**
This PR already uses refresh_token, because access_tokens live only for one hour (problematic for development).

### CODE REVIEW

File: GrandCentralBoard/GoogleAPIDataProvider.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        if accessToken == nil {
+            tokenProvider.accessTokenFromRefreshToken({ [weak self] response in
+                switch response {
+                case .Failure:
+                    completion(.Failure(.AuthorizationError))
+                case .Success(let newToken):
+                    self?.accessToken = newToken
+                    callRequest()
```
Comment (michallaskowski) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: Question: How to handle tokens? Access token may become invalid, and we know it after the request. There is also a time to live specified when getting one (but not sure if we can rely on that). Maybe there is an elegant way to do this (operations, RxSwift). WDYT?

File: GrandCentralBoard/GoogleTokenProvider.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//
+
+import Alamofire
+
+enum GoogleRefreshTokenError: ErrorType {
+    case FailedResponseParsing
+    case UnderlyingError(NSError)
+}
+
+class GoogleTokenProvider {
```
Comment (ochococo) [inline GrandCentralBoard/GoogleTokenProvider.swift]: `final` ?

File: GrandCentralBoard/GoogleAPIDataProvider.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+//
+
+import Alamofire
+
+enum GoogleAPIDataError: ErrorType {
+    case AuthorizationError
+    case UnderlyingError(NSError)
+}
+
+class GoogleAPIDataProvider {
```
Comment (ochococo) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: `final`?

File: GrandCentralBoard/GoogleAPIDataProvider.swift (hunk 1 above)
Comment (ochococo) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: You could create a queue and retain the requests, then dequeue it after the `accessToken` is received.

File: GrandCentralBoard/GoogleAPIDataProvider.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+//  GoogleAPIDataProvider.swift
+//  GrandCentralBoard
+//
+//  Created by Michał Laskowski on 06.04.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Alamofire
+
+enum GoogleAPIDataError: ErrorType {
```
Comment (ochococo) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: May I suggest a slight formatting improvement:

Type: `let int: Int`
Inheritance: `Error : ErrorType`

Comment (michallaskowski) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: Sure, but even Apple uses the same format (first one) for inheritance (https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Inheritance.html), and I think also Swiftlint would argue if we would add it to the project.

File: GrandCentralBoard/GoogleTokenProvider.swift [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        Alamofire.request(.POST, googleRefreshTokenURL, parameters: [
+            "client_id": clientID,
+            "client_secret": clientSecret,
+            "refresh_token": refreshToken,
+            "grant_type": "refresh_token"
+            ]).responseJSON { response in
+                switch response.result {
+                case .Failure(let error): completion(.Failure(.UnderlyingError(error)))
+                case .Success(let json):
+                    if let token = (json as? NSDictionary)?.valueForKey("access_token") as? String {
```
Comment (ochococo) [inline GrandCentralBoard/GoogleTokenProvider.swift]: We can use `Decodable` here.

File: GrandCentralBoard/GoogleAPIDataProvider.swift (hunk 4 above)
Comment (ochococo) [inline GrandCentralBoard/GoogleAPIDataProvider.swift]: See how the rest of the project is formatted. It's a suggestion though. Up to you.

Btw. Apple made this once:

1cea928

**Comment by michallaskowski** [comment]
Closing, will reopen a new one referencing GCB-128

**Pull request closed without merging**